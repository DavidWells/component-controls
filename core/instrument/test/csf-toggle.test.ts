import { loadStoriesTests } from './loadTestFiles';

describe('csf-toggle', () => {
  loadStoriesTests('csf', 'toggle');
});
