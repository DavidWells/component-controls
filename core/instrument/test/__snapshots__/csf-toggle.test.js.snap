// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csf-toggle Toggle story 1`] = `
Object {
  "properties": Array [
    Object {
      "loc": SourceLocation {
        "end": Position {
          "column": 32,
          "line": 6,
        },
        "start": Position {
          "column": 6,
          "line": 6,
        },
      },
      "name": "title",
      "value": "Components/Toggle",
    },
    Object {
      "loc": SourceLocation {
        "end": Position {
          "column": 23,
          "line": 7,
        },
        "start": Position {
          "column": 6,
          "line": 7,
        },
      },
      "name": "component",
      "value": "Toggle",
    },
  ],
  "stories": Object {
    "allToggles": Object {
      "arguments": Array [],
      "location": Object {
        "end": Object {
          "column": 5,
          "line": 29,
        },
        "start": Object {
          "column": 30,
          "line": 10,
        },
      },
      "name": "allToggles",
      "source": "() => {
      const [checked, setChecked] = React.useState(false);
      return (
        <div>
          <p>Default toggle</p>
          <Toggle checked={checked} onChange={check => setChecked(check)} />
          <br />
          <p>Custom labels</p>
          <Toggle
            checked={checked}
            labels={{
              true: 'YES',
              false: 'NO!',
            }}
            onChange={check => setChecked(check)}
          />
          <br />
        </div>
      );
    }",
    },
  },
  "title": "Components/Toggle",
}
`;
